### PR TITLE
[easy] Address some compilation warnings

### PR DESCRIPTION
**Pull request by dentiny**
Met some warnings during compilation, this PR fix them all.
Most of the warning here are caused by comparison between signed and unsigned integers.

### CODE REVIEW

File: src/core/parser/duckpgq_parser.cpp [hunk 1]
```diff
@@ -22,7 +22,6 @@ namespace core {
 
 ParserExtensionParseResult duckpgq_parse(ParserExtensionInfo *info,
                                          const std::string &query) {
-  auto parse_info = (DuckPGQParserExtensionInfo &)(info);
```
Comment (dentiny) [inline src/core/parser/duckpgq_parser.cpp:25]: Variable not used anywhere.

File: src/core/functions/scalar/local_clustering_coefficient.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@ static void LocalClusteringCoefficientFunction(DataChunk &args,
 
   DuckPGQBitmap neighbors(v_size);
 
-  for (int32_t n = 0; n < args.size(); n++) {
+  for (idx_t n = 0; n < args.size(); n++) {
```
Comment (dentiny) [inline src/core/functions/scalar/local_clustering_coefficient.cpp:45]:
```
/home/ubuntu/duckpgq-extension/src/core/functions/scalar/local_clustering_coefficient.cpp:45:25: warning: comparison of integer expressions of different signedness: ‘int32_t’ {aka ‘int’} and ‘duckdb::idx_t’ {aka ‘long unsigned int’} [-Wsign-compare]
   45 |   for (int32_t n = 0; n < args.size(); n++) {
      |                       ~~^~~~~~~~~~~~~
```

File: src/core/functions/scalar/local_clustering_coefficient.cpp [hunk 3]
```diff
@@ -54,15 +54,15 @@ static void LocalClusteringCoefficientFunction(DataChunk &args,
       continue;
     }
     neighbors.reset();
-    for (size_t offset = v[src_node]; offset < v[src_node + 1]; offset++) {
+    for (int64_t offset = v[src_node]; offset < v[src_node + 1]; offset++) {
```
Comment (dentiny) [inline src/core/functions/scalar/local_clustering_coefficient.cpp:57]:
```
**/home/ubuntu/duckpgq-extension/src/core/functions/scalar/local_clustering_coefficient.cpp:57:46: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int64_t’ {aka ‘long int’} [-Wsign-compare]
   57 |     for (size_t offset = v[src_node]; offset < v[src_node + 1]; offset++) {
      |                                       ~~~~~~~^~~~~~~~~~~~~~~~~**
```

**Comment by Dtenwolde** [comment]
Thanks for the PR!
Regarding the DuckPGQParserExtensionInfo, I don't see any use for it in the repo. I think it is an artefact of the early days of the extension when I wasn't quite sure if we needed to use this. Since there are no use cases, I think this entire struct should be removed to avoid having redundant code in the repo.

**Comment by dentiny** [comment]
Searching through the codebase, it's actually used in one place:
https://github.com/cwida/duckpgq-extension/blob/fb38a09818121021c268b4328f9fb82500f233d0/src/include/duckpgq/core/parser/duckpgq_parser.hpp#L34

which seems be a necessary field for `ParserExtension`: https://github.com/cwida/duckdb-pgq/blob/0c82f0ed582ed2737216539845028826badb4bda/src/include/duckdb/parser/parser_extension.hpp#L82-L94

Would like to double confirm if it's completely removable.

**Comment by Dtenwolde** [comment]
I commented out the struct and any uses, and all tests passed; however, maybe it's best to just leave it in anyway. Sorry for the confusion

**Review by Dtenwolde — APPROVED** [review]

**Comment by dentiny** [comment]
Thank you for spending time taking a look for me!

**Pull request merged**